lib/vrl/web-playground/public/vrl-highlighter.js

```diff
@@ -0,0 +1,636 @@
+            [
+                /([a-zA-Z_!]+)(\()/,
+                {
+                    cases: {
+                        $2: ['keyword', '']
+                    }
+                    //   log: 'in function invoke::found::\n\n$0'
+                },
+            ],
+
+            // rstrings
+            [/r'[^']+'/, { token: "regexp" /*log: 'root_r_string::\n\n$0'*/ }],
+
+            // timestamps
+            [/t'[^']+'/, { token: "regexp" /*log: 'function_arg_r_string::\n\n$0'*/ }],
+
+            // field access, eg: .foo
+            [/(\.[^\ \=]+)([\ |\=])/, {
+                cases: {
+                    $2: ["entity", ""]
+                }
+            }],
+
+            // sstrings
+            [/s'[^']+'/, { token: "string" /*log: 'root_s_string::\n\n$0'*/ }],
+
+            // identifiers and keywords
+            [/\@[a-zA-Z_]\w*/, "variable.predefined"],
+
+            [
+                /[a-zA-Z_]\w*/,
+                {
+                    cases: {
+                    this: "variable.predefined",
+                    "@keywords": { token: "keyword.$0" },
+
+                    "@default": "",
+                    },
+                },
+            ],
+
+            // whitespace
+            [/[ \t\r\n]+/, ""],
+
+            // Comments
+            [/###/, "comment", "@comment"],
+            [/#.*$/, "comment"],
+
+            // Function invokes
+
+            // regular expressions
+            ["///", { token: "regexp", next: "@hereregexp" }],
+
+            [/^(\s*)(@regEx)/, ["", "regexp"]],
+            [/(\()(\s*)(@regEx)/, ["@brackets", "", "regexp"]],
+            [/(\,)(\s*)(@regEx)/, ["delimiter", "", "regexp"]],
+            [/(\=)(\s*)(@regEx)/, ["delimiter", "", "regexp"]],
+            [/(\:)(\s*)(@regEx)/, ["delimiter", "", "regexp"]],
+            [/(\[)(\s*)(@regEx)/, ["@brackets", "", "regexp"]],
+            [/(\!)(\s*)(@regEx)/, ["delimiter", "", "regexp"]],
+            [/(\&)(\s*)(@regEx)/, ["delimiter", "", "regexp"]],
+            [/(\|)(\s*)(@regEx)/, ["delimiter", "", "regexp"]],
+            [/(\?)(\s*)(@regEx)/, ["delimiter", "", "regexp"]],
+            [/(\{)(\s*)(@regEx)/, ["@brackets", "", "regexp"]],
+            [/(\;)(\s*)(@regEx)/, ["", "", "regexp"]],
+
+            // delimiters
+            [
+                /}/,
+                {
+                    cases: {
+                    "$S2==interpolatedstring": {
+                        token: "string",
+                        next: "@pop",
+                    },
+                    "@default": "@brackets",
+                    },
+                },
+            ],
+            [/[{}()\[\]]/, "@brackets"],
+            [/@symbols/, "delimiter"],
+
+            // numbers
+            [/\d+[eE]([\-+]?\d+)?/, "number.float"],
+            [/\d+\.\d+([eE][\-+]?\d+)?/, "number.float"],
+            [/0[xX][0-9a-fA-F]+/, "number.hex"],
+            [/0[0-7]+(?!\d)/, "number.octal"],
+            [/\d+/, "number"],
```

Comment:
There's a bunch of stuff here that we don't support, and some things that we do support, that is still missing.

Here's an example of what we support:

https://deploy-preview-15557--vrl-playground.netlify.app/?state=eyJwcm9ncmFtIjoiMFxuOVxuLTEwXG4xMDBfMDAwXG4tMTBfMDAwXzAwMFxuMTAuXG4xMC4yXG4wLjEyXG4tMTIuMzQ1XzY3OCIsImV2ZW50Ijp7Im1lc3NhZ2UiOiJIZWxsbyBWUkwiLCJmb28iOiJkZWxldGUgbWUiLCJodHRwX3N0YXR1cyI6IjIwMCJ9LCJpc19qc29ubCI6ZmFsc2V9